packages/react-router-dom/index.tsx

```diff
@@ -725,12 +733,10 @@ export function RouterProvider({
                 location={state.location}
                 navigationType={state.historyAction}
                 navigator={navigator}
-                future={{
-                  v7_relativeSplatPath: router.future.v7_relativeSplatPath,
-                }}
+                future={routerFuture}
               >
                 {state.initialized || router.future.v7_partialHydration ? (
-                  <DataRoutes
+                  <MemoizedDataRoutes
```

Comment:
Has to `React.memo` this in order to avoid unnecessary re-renders of route components when we updated `ViewTransitionContext`.

I would have thought I'd have needed to memoize `Router` as well but it doesn't appear that one needs to be changed